pkg/api/vmtemplate/store.go

```diff
@@ -0,0 +1,119 @@
+package vmtemplate
+
+import (
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/vm/pkg/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/rancher/apiserver/pkg/types"
+	ctlvmv1alpha1 "github.com/rancher/vm/pkg/generated/controllers/vm.cattle.io/v1alpha1"
+	"github.com/rancher/vm/pkg/ref"
+)
+
+const (
+	vmTemplateLabel = "vmtemplate.vm.cattle.io/vmTemplateID"
+)
+
+type wrapStore struct {
+	types.Store
+	vmTemplateCache        ctlvmv1alpha1.VMTemplateCache
+	vmTemplateVersions     ctlvmv1alpha1.VMTemplateVersionClient
+	vmTemplateVersionCache ctlvmv1alpha1.VMTemplateVersionCache
+	mu                     sync.Mutex
+}
+
+func (s *wrapStore) Create(request *types.APIRequest, schema *types.APISchema, data types.APIObject) (types.APIObject, error) {
+	ns := data.Data().String("metadata", "namespace")
+	if ns == "" {
+		data.Data().SetNested(config.Namespace, "metadata", "namespace")
+	}
```

Comment:
Use namespaceStore for this

Reply:
namespaceStore for what? create the namespace if it doesn't exist?

Reply:
https://github.com/rancher/harvester/blob/master/pkg/api/store/namespace.go